x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -8604,6 +8604,80 @@ public void testMultipleKnnQueriesInPrefilters() {
         assertTrue(secondKnnFilters.contains(firstOr.right()));
     }
 
+    public void testKnnImplicitLimit() {
```

Comment:
can we get a test when knn is used before stats? e.g. `from test | where knn(...) | stats x = COUNT(*)`.
I know it might seem like this does not make a lot of sense to use STATS after KNN, but it would be good to check what `k` do we set.
since `k` is set through an optimization rule, it would be great to also have CSV tests with knn used before STATS, RERANK etc.

Reply:
I added unit tests as part of https://github.com/elastic/elasticsearch/commit/9d3c85f4fd73f55dc737d1d1ad965bc7d7a6f821, and added CSV tests in af3296c84776ffeb36d4b25ed058d094a904c47a